### PR TITLE
Split /trillium/longTermCare (adding additional page)

### PR DESCRIPTION
**Pull request by nmakuch**
### `/trillium/longTermCare` -> `/trillium/longTermCare/type` -> `/trillium/longTermCare/type/amount`
- This PR resolves [github issue 258](https://github.com/cds-snc/cra-claim-tax-benefits/issues/258)
- This fixes the issue of having a double barreled question on this page
- Tests were updated to accept the new page headers
- Unit tests were added for the new page

### GIF
temp2

### CODE REVIEW

File: cypress/fixtures/checkAnswersRows.json [hunk 1]
```diff
@@ -53,11 +53,7 @@
       "value": "No"
     },
     {
-      "key": "Did you pay home energy costs on a reserve?",
-      "value": "No"
```
Comment (pcraig3) [inline cypress/fixtures/checkAnswersRows.json]: I would think we still want to keep this.

File: locales/fr.json [hunk 2]
```diff
@@ -401,26 +402,7 @@
...
-  "Kingston": "Kingston",
-  "Kitchener-Cambridge-Waterloo": "Kitchener-Cambridge-Waterloo",
-  "London": "London",
-  "Oshawa": "Oshawa",
-  "Ottawa": "Ottawa",
-  "Peterborough": "Peterborough",
-  "St. Catharines-Niagara": "St. Catharines-Niagara",
-  "Sudbury": "Sudbury",
-  "Thunder Bay": "Thunder Bay",
-  "Windsor": "Windsor",
```
Comment (pcraig3) [inline locales/fr.json]: Are these actually deleted or no? I thought we had a big list now (on the Climate Action incentive page)

File: locales/fr.json [hunk 3]
```diff
@@ -463,5 +445,33 @@
...
+  "Kingston": "Kingston",
+  "Kitchener-Cambridge-Waterloo": "Kitchener-Cambridge-Waterloo",
+  "London": "London",
+  "Oshawa": "Oshawa",
+  "Ottawa": "Ottawa",
+  "Peterborough": "Peterborough",
+  "St. Catharines-Niagara": "St. Catharines-Niagara",
+  "Sudbury": "Sudbury",
+  "Thunder Bay": "Thunder Bay",
+  "Windsor": "Windsor",
```
Comment (pcraig3) [inline locales/fr.json]: Ah, nvm.

File: routes/deductions/deductions.controller.js [hunk 4]
```diff
@@ -215,15 +216,31 @@ module.exports = function(app) {
...
+  app.get('/trillium/longTermCare/type', renderWithData('deductions/trillium-longTermCare-type'))
+  app.post(
+    '/trillium/longTermCare/type',
+    checkSchema(trilliumLongTermCareTypeSchema),
+    checkErrors('deductions/trillium-longTermCare-type'),
+    doYesNo('trilliumLongTermCareTypeClaim', 'trilliumLongTermCareAmount'),
+    // These only apply if the user clicked "no"
+    // If they clicked "Yes", they will be redirected by `doYesNo()`
+    (req, res, next) => {
+      req.session.deductions.trilliumEnergyAmount = 0
```
Comment (pcraig3) [inline routes/deductions/deductions.controller.js]: This should be `req.session.deductions.trilliumEnergyAmount = 0`

File: routes/deductions/deductions.controller.js [hunk 5]
```diff
@@ -273,3 +290,27 @@ const postEnergyReserve = (req, res, next) => {
...
+
+    return res.redirect('/deductions/climate-action-incentive')
+  }
+
+  if (req.query.ref && req.query.ref === 'checkAnswers') {
+    return returnToCheckAnswers(req, res, true)
+  }
+
+  next()
+}
```
Comment (pcraig3) [inline routes/deductions/deductions.controller.js]: might want to clean this up one day, but it works 🤷‍♂

File: routes/deductions/deductions.controller.js [hunk 6]
```diff
@@ -215,15 +216,31 @@ module.exports = function(app) {
...
+  )
+
   app.get(
-    '/trillium/longTermCare/amount',
+    '/trillium/longTermCare/type/amount',
     renderWithData('deductions/trillium-longTermCare-amount'),
   )
   app.post(
-    '/trillium/longTermCare/amount',
+    '/trillium/longTermCare/type/amount',
```
Comment (pcraig3) [inline routes/deductions/deductions.controller.js]: We're changing the URL here because of 'doYesNo', right?

Comment (nmakuch) [reply routes/deductions/deductions.controller.js]: yeah exactly

File: cypress/fixtures/checkAnswersRows.json (hunk 1 above)
Comment (nmakuch) [reply cypress/fixtures/checkAnswersRows.json]: It is now, 'Did you live on a reserve?'

Comment (nmakuch) [reply cypress/fixtures/checkAnswersRows.json]: should they both be in there?

**Pull request merged**